core/service/transaction/register.ts

```diff
@@ -0,0 +1,91 @@
+import { ITransactionService } from 'core/service/transaction';
+import { IAssetRegister, Transaction } from 'shared/model/transaction';
+import { Account } from 'shared/model/account';
+import Response from 'shared/model/response';
+import AccountRepo from 'core/repository/account';
+import config from 'shared/util/config';
+import BUFFER from 'core/util/buffer';
+import {ITableObject} from 'core/util/common';
+
+class TransactionRegisterService implements ITransactionService<IAssetRegister> {
+
+    async create(trs: Transaction<IAssetRegister>, data: IAssetRegister ): Promise<IAssetRegister> {
+        trs.recipientAddress = null;
+        return {
+            referral: data.referral
+        };
+    }
+
+    getBytes(trs: Transaction<IAssetRegister>): Buffer {
+        const buff = Buffer.alloc(BUFFER.LENGTH.INT64);
+        BUFFER.writeUInt64LE(buff, trs.asset.referral, 0);
+        return buff;
+    }
+
+    async verifyUnconfirmed(trs: Transaction<IAssetRegister>, sender: Account): Promise<Response<void>> {
+        return new Response();
+    }
+
+    async verify(trs: Transaction<IAssetRegister>, sender: Account): Promise<Response<void>> {
+        const errors = [];
+
+        if (!trs.asset.referral) {
+            errors.push('Missing referral');
+        }
+
+        if (sender) {
```

Comment:
you always get account for sender. Previously, in BlockService you have `getOrCreateAccount` method, which will provide you account.